.pfnci/matrix.yaml

```diff
@@ -286,8 +286,8 @@
   cutensor: "1.7"
   cusparselt: "0.2.0"
   cudnn: "8.8"
-  python: "3.11"
-  numpy: "1.24"
+  python: "3.12"
+  numpy: "1.26"
   scipy: "1.10"
```

Comment:
SciPy 1.10 does not provide binary package for Python 3.12. I guess it would be better to add Python 3.12 axis after supporting SciPy 1.11?